docs/spec.md

```diff
@@ -16,16 +37,15 @@ size 12345
 (ending \n)
 ```
 
-The pointer file should be small (less than 200 bytes), and consist of only
-ASCII characters.  Libraries that generate this should write the file
-identically, so that different implementations write consistent pointers that
-translate to the same Git blob OID.  This means:
+For testing compliance of any tool generating its own pointer files, the
+reference is this official Git LFS tool:
+
+NOTE: exact pointer command behavior TBD!
 
-* Use properties "version", "oid", and "size" in that order.
-* Separate the property from its value with a single space.
-* Oid has a "sha256:" prefix.  No other hashing methods are currently supported
-for Git LFS oids.
-* Size is in bytes.
+* Run `git lfs pointer` to generate a pointer file for the given local file.
+* Run `git lfs pointer` to compare the blob OID of the generated pointer files.
+* Tools that parse and regenerate pointer files MUST preserve keys that they
+don't know or care about.
 
 Note: Earlier versions only contained the OID, with a `# comment` above it.
```

Comment:
If this spec is supposed to indicate how to parse older versions then it
would need one section per version, and each section should be self
contained explaining exactly what version it is, how to identify it from
the header, and of course what the format is, and at least one example file
pointer.

It would be the best way to let third parties write correct tools.
On Apr 20, 2015 12:55 PM, "risk danger olson" <notifications@github.com>
wrote:

> In docs/spec.md
> <https://github.com/github/git-lfs/pull/246#discussion_r28722784>:
>
> >
> > -* Use properties "version", "oid", and "size" in that order.
> > -* Separate the property from its value with a single space.
> > -* Oid has a "sha256:" prefix.  No other hashing methods are currently supported
> > -for Git LFS oids.
> > -* Size is in bytes.
> > +* Run `git lfs pointer` to generate a pointer file for the given local file.
> > +* Run `git lfs pointer` to compare the blob OID of the generated pointer files.
> > +* Tools that parse and regenerate pointer files MUST preserve keys that they
> > +don't know or care about.
> >
> >  Note: Earlier versions only contained the OID, with a `# comment` above it.
>
> Sure, then Git LFS tools don't know how to parse older pointer files. I
> was thinking of removing this section though, because the chance of anyone
> besides me seeing these old blobs is pretty minimal.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/github/git-lfs/pull/246/files#r28722784>.
>